frontend/src/metabase/nav/containers/MainNavbar/SidebarItems/SidebarOnboardingSection/SidebarOnboardingSection.tsx

```diff
@@ -94,29 +92,82 @@ export function SidebarOnboardingSection({
 
   const isMobileSafe = useMediaQuery(`(min-width: ${breakpoints.sm})`);
 
+  const canAddDatabase = isAdmin;
+
+  /**
+   * the user must have:
+   *   - "write" permissions for the root collection AND
+   *   - either:
+   *       a) !uploadsEnabled => data access to any of the databases OR
+   *       b) uploadsEnabled => "upload" permissions for the database for which uploads are enabled
+   */
+  const isUploadEnabled = !!uploadDbId;
+  const rootCollection = collections.find(
+    c => c.id === "root" || c.id === null,
+  );
+  const canCurateRootCollection = rootCollection?.can_write;
+  const canUploadToDatabase = databases
+    ?.find(db => db.id === uploadDbId)
+    ?.canUpload();
+  const canUpload =
+    canCurateRootCollection &&
+    (isUploadEnabled ? canUploadToDatabase : hasDataAccess);
+
+  function AddDatabaseButton() {
```

Comment:
can AddDatabaseButton and UploadSpreadsheetButton get pulled out of SidebarOnboardingSection?

Reply:
ABSOlutely!
Do we allow the `/components` folder inside already nested folder?

For example, I'd put them in `frontend/src/metabase/nav/containers/MainNavbar/SidebarItems/SidebarOnboardingSection/components/AddDatbaseButton.tsx`

Reply:
I personally wouldn't be bothered if they were in the same file, just moved out of the render method. What you're suggesting I'm not seeing us do anywhere, but I'm not offended by it.